sphinx/sphinx.gradle

```diff
@@ -8,12 +8,22 @@ description = 'Generates docs for the python libraries provided in Deephaven Cor
 
 evaluationDependsOn ':Integrations'
 
+tasks.register('copyPyClientWhl', Copy) {
+    dependsOn ':pyclient:buildPyClient'
+    from "$rootDir/pyclient/build/wheel/"
+    into layout.buildDirectory.file('sphinx-image/wheel/')
+}
+
 def sphinxDockerfile = tasks.register('sphinxDockerfile', Dockerfile) {
+    dependsOn 'copyPyClientWhl'
```

Comment:
this should be in the sphinx image, it doesn't need to be here, and should be an input.files instead of just a dependsOn, roughly line 32 or so should get this:

```
inputs.files tasks.named('copyPyClientWhl').get().outputs.files
```

Moving this would let the dockerfile task be left if up to date, and ensure only the image gets rebuilt (this has also no measurable benefit, but a few ms here and a few ms there...)

Preferring the inputs... and ...outputs... lets gradle know why we depend on that task, and so if the upstream task re-runs but actually has no changes that we care about, this downstream task can avoid re-running. That is more of a win, since we get to avoid talking to docker, and risking re-running even more downstream work.